filters/sed/editor.go

```diff
@@ -0,0 +1,474 @@
+package sed
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"regexp"
+)
+
+// TODO:
+// - we need to document that only non-zero matches are replaced
+// - we can document that it is currently not supported, together
+//   with the templating
+
+const (
+	readBufferSize             = 8 << 10
+	defaultMaxEditorBufferSize = 2 << 20
```

Comment:
I am not so into bit shifting, if it's not necessary for performance.
I like much more:
readBufferSize = 8192
defaultMaxEditorBufferSize = 256 * readBufferSize

I guess that defaultMaxEditorBufferSize and readBufferSize are dependent on each other makes it also more clear.
If you really like `8 << 10`, then please use `defaultMaxEditorBufferSize = 256 * readBufferSize`.

Reply:
i picked up this notation in the Go standard library, where I seen it used as:

`<< 10` means Ki
`<< 20` means Mi
`<< 30` means Gi

So the meaning of these values should be 8KiB and 2MiB. I'll replace them though if it is really not intuitive.

Reply:
Calculating one from the other doesn't make too much sense in my view, because these are not related values.

Reply:
It makes sense, because you read x times readBufferSize and it would be weird to be able to read only 3.43 times readBufferSize as max.
This is what I am trying to argue for. :)